libRALFit/doc/Fortran/howtouse.rst

```diff
@@ -277,27 +281,162 @@ subroutine must implement the following interface:
            double precision, dimension(n), intent(in)  :: x
            double precision, dimension(n), intent(in)  :: y
            double precision, dimension(n*m), intent(out) :: HP
-           class(params_base_type), intent(in) :: params
+           class(params_base_type), intent(inout) :: params
          end subroutine eval_HP_type
     end interface
     :language: fortran
 
 .. f:subroutine:: eval_HP(status,n,m,x,y,HP,params)
 
    :p integer status [inout]: |eval_HP_status|
-			   
+
    :p integer n [in]: |eval_HP_n|
 
    :p integer m [in]: |eval_HP_m|
-		      
+
    :p real x(n) [in]: |eval_HP_x|
 
    :p real y(n) [in]: |eval_HP_y|
 
    :p real HP(n*m) [out]: |eval_HP_HP|
-			   
-   :p params_base_type params [in]: |eval_HP_params|
-				 
+
+   :p params_base_type params [inout]: |eval_HP_params|
+
+.. _fd:
+
+Checking and approximating derivatives
+--------------------------------------
+
+Verifying user-supplied first derivatives matrix
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One of the most common problems while trying to train a model is having wrong derivatives.
+Writing the derivative call-back function is error-prone and to address this, a derivative
+checker can be activated (set option ``check_derivatives``) for checking the
+derivatives provided by the call-back that evaluates :math:`J=\nabla r(x)`.
+The checker produces a table similar to
+
```

Comment:
Import non language-specific blocks of text from a macro to avoid having getting descriptions out of sync